stdlib/src/builtin/string.mojo

```diff
@@ -52,6 +52,23 @@ fn _ctlz(val: SIMD) -> __type_of(val):
     )
 
 
+@always_inline
+fn _str_compare(str1: String, str2: String) -> Int:
```

Comment:
Can you please provide a summary of what the logic is in the docstring? Also, please document that this can return {-1, 0, 1}.